File: Emulator/src/main/content/global/skill/slayer/npc/BasiliskNPC.java
package content.global.skill.slayer.npc;

import content.global.skill.slayer.items.MirrorShieldHandler;
import core.game.node.entity.combat.BattleState;
import core.game.node.entity.combat.CombatSwingHandler;
import core.game.node.entity.npc.AbstractNPC;
import core.game.world.map.Location;
import core.plugin.Initializable;
import org.rs.consts.NPCs;

/**
 * The type Basilisk npc.
 */
@Initializable
public class BasiliskNPC extends AbstractNPC {

    /**
     * Instantiates a new Basilisk npc.
     */
    public BasiliskNPC() {
        super(0, null);
    }

    /**
     * Instantiates a new Basilisk npc.
     *
     * @param id       the id
     * @param location the location
     */
    public BasiliskNPC(int id, Location location) {
        super(id, location);
    }

    @Override
    public AbstractNPC construct(int id, Location location, Object... objects) {
        return new BasiliskNPC(id, location);
    }

    @Override
    public CombatSwingHandler getSwingHandler(boolean swing) {
        return MirrorShieldHandler.INSTANCE;
    }

    @Override
    public void checkImpact(BattleState state) {
        super.checkImpact(state);
        MirrorShieldHandler.INSTANCE.checkImpact(state);
    }

    @Override
    public int[] getIds() {
        return new int[] { NPCs.BASILISK_1616, NPCs.BASILISK_1617 };
    }
}

File: game-server/src/main/content/global/skill/slayer/npc/BasiliskNPC.kt
package content.global.skill.slayer.npc

import content.global.skill.slayer.Tasks
import content.global.skill.slayer.items.MirrorShieldHandler
import core.game.node.entity.combat.BattleState
import core.game.node.entity.combat.CombatSwingHandler
import core.game.node.entity.npc.AbstractNPC
import core.game.world.map.Location
import core.plugin.Initializable

/**
 * The type Basilisk npc.
 */
@Initializable
class BasiliskNPC : AbstractNPC {
    /**
     * Instantiates a new Basilisk npc.
     */
    constructor() : super(0, null)

    /**
     * Instantiates a new Basilisk npc.
     *
     * @param id       the id
     * @param location the location
     */
    constructor(id: Int, location: Location?) : super(id, location)

    override fun construct(id: Int, location: Location, vararg objects: Any): AbstractNPC {
        return BasiliskNPC(id, location)
    }

    override fun getSwingHandler(swing: Boolean): CombatSwingHandler {
        return MirrorShieldHandler
    }

    override fun checkImpact(state: BattleState) {
        super.checkImpact(state)
        MirrorShieldHandler.checkImpact(state)
    }

    override fun getIds(): IntArray {
        return Tasks.BASILISKS.npcs
    }
}
